### PR TITLE
Fix to prevent 'undefined is not a function' like errors

### PR DESCRIPTION
**Pull request by galberola**
## TL;DR: 
If we don't do all the exercises in the right order, some might fail (`undefined is not a function`) due to the missing definition of 
`Array.prototype.concatAll()` and/or `Array.prototype.concatMap(function)`

## Problem
`Array.prototype.concatAll()` & `Array.prototype.concatMap(function)` are being defined in some of the exercises and added to `Array.prototype`.
For such they stay defined for the remaining time of the session.

The problem is that if we close the tab and get back in again, the session is resumed to the latest exercise completed, but this two definitions are non existent.

## Solution
We added at the beginning of each _verifier_ function a call to a `preVerifierHook` that adds the definition for `Array.prototype.concatAll()` & `Array.prototype.concatMap(function)`

### CODE REVIEW

**Comment by galberola** [comment]
This PR will also fixes #41 #26

**Comment by galberola** [comment]
This PR will also fix #41 #39

**Comment by morenoh149** [comment]
issue https://github.com/jhusain/learnrx/issues/26 says this PR referenced that issue. Could you elaborate on it's relation on issue #26
